### PR TITLE
Refine logging in pkg/selector/generic/namespace

### LINKED ISSUE #2886: Refine logging in `pkg/selector/generic/namespace`, refactor these function as method of a Struct, then move package level variable `log` as a field
> Please leave comments with your idea if you are interested in this issue. And comment `/assign` to operate self-assign before you work on it.

### PR DESCRIPTION
**Pull request by DaivikDave**
Signed-off-by: Daivik <daivikdave229@gmail.com>

### What problem does this PR solve?
Close #2886 

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

CHANGELOG

- [ ] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
@DaivikDave Thanks for your contribution! Please update the `CHANGELOG`

**Comment by DaivikDave** [comment]
Thanks @WangXiangUSTC , Updated !

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

File: pkg/selector/pod/selector.go [hunk 1]
```diff
@@ -167,7 +170,7 @@ func selectSpecifiedPods(ctx context.Context, c client.Client, spec v1alpha1.Pod
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply pkg/selector/pod/selector.go]: ignore

**Review by cwen0 — APPROVED** [review]
LGTM

**Pull request merged**